src/WebJobs.Script/Description/ScriptFunctionInvoker.cs

```diff
@@ -189,29 +181,56 @@ private void InitializeEnvironmentVariables(Dictionary<string, string> environme
             }
         }
 
-        private async Task ProcessInputBindingsAsync(string functionInstanceOutputPath, IBinder binder, Dictionary<string, string> bindingData, Dictionary<string, string> environmentVariables)
+        private async Task ProcessInputBindingsAsync(object input, string functionInstanceOutputPath, IBinder binder, Dictionary<string, string> bindingData, Dictionary<string, string> environmentVariables)
         {
             // if there are any input or output bindings declared, set up the temporary
             // output directory
-            var nonTriggerInputBindings = _inputBindings.Where(p => !p.IsTrigger);
-            if (_outputBindings.Count > 0 || nonTriggerInputBindings.Any())
+            if (_outputBindings.Count > 0 || _inputBindings.Any())
             {
                 Directory.CreateDirectory(functionInstanceOutputPath);
             }
 
             // process input bindings
-            foreach (var inputBinding in nonTriggerInputBindings)
+            foreach (var inputBinding in _inputBindings)
             {
                 string filePath = System.IO.Path.Combine(functionInstanceOutputPath, inputBinding.Name);
                 using (FileStream stream = File.OpenWrite(filePath))
                 {
-                    BindingContext bindingContext = new BindingContext
+                    // If this is the trigger input, write it directly to the stream.
+                    // The trigger binding is a special case because it is early bound
+                    // rather than late bound as is the case with all the other input
+                    // bindings.
+                    if (inputBinding.IsTrigger)
+                    {
+                        if (input is string)
+                        {
+                            using (StreamWriter sw = new StreamWriter(stream))
+                            {
+                                await sw.WriteAsync((string)input);
+                            }
+                        }
+                        else if (input is byte[])
+                        {
+                            byte[] bytes = input as byte[];
+                            await stream.WriteAsync(bytes, 0, bytes.Length);
+                        }
+                        else if (input is Stream)
```

Comment:
I've changed the blob binding to always bind to Stream as you'll see below. Previously we were always converting to string, which doesn't work of course for the image scenario. Also binding to Stream is more efficient, since we never have to load the entire blob into memory - it streams into the temp file, and scripts can stream out of the temp file.